beater/config.go

```diff
@@ -1,15 +1,15 @@
 // Config is put into a different package to prevent cyclic imports in case
 // it is needed in several locations
 
-package config
+package beater
 
 import "github.com/elastic/beats/libbeat/common"
 
-type Config struct {
-	Host   string         `config:"host"`
-	Server *common.Config `config:"server"`
+type BeaterConfig struct {
+	Host            string         `config:"host"`
+	ApmServerConfig *common.Config `config:"server"`
 }
 
-var DefaultConfig = Config{
+var DefaultBeaterConfig = BeaterConfig{
```

Comment:
I was very confused by the several layers of configs we have. I know this is not as short as usual Golang idioms, but then I clearly know what kind of configuration is this.
This all PR in general comes out of experience trying to understand the code :)